src/components/WebLogView/WebLogView.tsx

```diff
@@ -21,6 +23,20 @@ export function WebLogView({
   onSelectRequest,
   onUpdateGroup,
 }: WebLogViewProps) {
+  const selectedRequest = useMemo(
+    () => requests.find((data) => data.id === selectedRequestId),
+    [requests, selectedRequestId]
+  )
+
+  // Sync selectedRequest when requests change to show updates in correlation preview
+  useDeepCompareEffect(() => {
+    if (!selectedRequest) {
+      return
+    }
+
+    onSelectRequest(selectedRequest)
+  }, [selectedRequest, onSelectRequest])
```

Comment:
When a request is clicked, we copy it to `selectedRequest` state. This is fine if request doesn't change, but when inspecting a request and editing a correlation rule the `requests` object will change and `selectedRequest` in parent component would become stale. This effect addresses it by syncing `selectedRequest` when `requests` change. It's not the ideal solution, but since it's likely the rule preview will change in the future, I didn't invest time in refactoring all consumers of `WebLogView`